worker/catacomb/catacomb.go

```diff
@@ -110,14 +110,30 @@ func Invoke(plan Plan) (err error) {
 	// This goroutine runs the work func and stops the catacomb with its error;
 	// and waits for for the listen goroutine and all added workers to complete
 	// before marking the catacomb's tomb Dead.
+	done := make(chan bool)
+	catacomb.tomb.Go(WaitTillDone(done))
 	go func() {
```

Comment:
given that WaitTillDone just exists to run until func() returns, it makes a lot more sense to do this as:
catacomb.tomb.Go(func() error {
  defer catacomb.wg.Wait()
  catacomb.Kill(...)
  return nil
})

Rather than creating a thunk and an extra goroutine that just exists to track the existing goroutine.